src/components/BarrierFree/BarrierFree.js

```diff
@@ -0,0 +1,177 @@
+import React, { useEffect } from 'react';
+import { useSelector } from 'react-redux';
+import { containsExtent } from 'ol/extent';
+import WKT from 'ol/format/WKT';
+import getFeatureGeometry from '../../utils/getFeatureGeometry';
+
+const focusClass = 'wkp-bf-focus';
+let tabFeature = false;
+let featLayer = null;
+let visibleLayers = [];
+let visibleFeatures = [];
+let featureIndex = -1;
+
+function BarrierFree() {
+  const zoom = useSelector(state => state.map.zoom);
+  const center = useSelector(state => state.map.center);
+  const resolution = useSelector(state => state.map.resolution);
+
+  const map = useSelector(state => state.app.map);
+  const layerService = useSelector(state => state.app.layerService);
+
+  // Filtering function based on generalization levels.
+  const wktFormat = new WKT();
+  const filterFeat = feature => {
+    const geometry = getFeatureGeometry(feature, resolution, wktFormat);
+
+    if (!geometry || feature.get('visibility') < resolution * 10) {
+      return false;
+    }
+    return true;
+  };
+
+  const removeHighlight = () => {
+    // Remove feature highlight
+    if (featLayer) {
+      featLayer.clickedFeature = null;
+      featLayer.olLayer.changed();
+    }
+    tabFeature = false;
+  };
+
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const setVisibleFeats = () => {
+    visibleLayers = layerService
+      .getLayersAsFlatArray()
+      .reverse()
+      .filter(l => !l.getIsBaseLayer() && l.getVisible());
+
+    const mapExtent = map.getView().calculateExtent(map.getSize());
+
+    let visFeats = [];
+    visibleLayers.forEach(l => {
+      if (l && l.olLayer) {
+        const features = l.olLayer.getSource().getFeatures();
+        if (features.length > 0) {
+          features.forEach(f => f.set('layerRef', l));
+          const visibleFeats = features.filter(
+            feat =>
+              containsExtent(mapExtent, feat.getGeometry().getExtent()) &&
```

Comment:
Use getFeaturesInExtent directly from the source. 
https://openlayers.org/en/master/apidoc/module-ol_source_Vector-VectorSource.html#getFeaturesInExtent